### PR TITLE
[A11y] Treat last character as 'end of buffer'

### LINKED ISSUE #6986: Properly handle and test "end of buffer" in UiaTextRange
When moving by unit, UiaTextRange keeps track of the last character to be the "end of the buffer". This provides significant performance improvements in our accessibility model for word navigation in particular.

However, it is still possible to create a UiaTextRange outside of that optimized buffer size. This has lead to a few bugs like #6402 and ##5243 across Terminal and ConHost.

This issue tracks taking the time to properly test this and polish this architectural change. Especially considering that this breaks a significant number of tests that already exist.

### LINKED ISSUE #7960: UIA: incorrect word boundary
### STR:
1. In a Linux directory with several files, run `ls | pr -2Tn`.
2. Navigate the output by word (with either NVDA or Narrator).

### Expected behaviour:
Each word is announced.

### Observed behaviour:
When navigating across a line boundary, strange things happen: it looks like the entire line (except the last character or two) are included in a word!

Cc @carlos-zamora.

### PR DESCRIPTION
**Pull request by carlos-zamora**
## Summary of the Pull Request
Updates our `UiaTextRange` to treat the last legible character as the end of the buffer. Doing so significantly improves performance of word navigation because screen readers no longer have to take into account the whitespace following the end of the prompt.

This also fixes #7960. This bug was caused by the `UiaTextRange` occasionally having its `_blockRange` member set to `true`. As a result, the screen reader would read the block range of text from "start" to "end" as if it was a block selection created by Alt+Mouse.

## References
#7000 - Epic
Closes #6986 
Closes #7960 

## Detailed Description of the Pull Request / Additional comments
_Coming soon!_

## Validation Steps Performed
- [ ] Tests added/passed
- [x] NVDA - manual testing
- [x] Accessibility Insights - manual testing
- [ ] Narrator - manual testing

### CODE REVIEW

**Comment by carlos-zamora** [comment]
## TODO
- [ ] Bug: "move to next word" when on the last word does not tell screen reader that we're on the bottom
- [ ] Fix tests
- [ ] Get performance traces of word navigation before/after this change

**Comment by carlos-zamora** [comment]
Most of the code that's submitted won't be touched, so a review would be much appreciated!

**Comment by codeofdusk** [comment]
Bug:
1. Start NVDA and conhost.
2. type "ssh" followed by a space.
3. Delete the entered text with backspace.

NVDA says, on each press of backspace: "h, h, s, s".

NVDA should say (and said before this PR): "space, h, s, s".

File: src/buffer/out/textBuffer.cpp [hunk 1]
```diff
@@ -1237,40 +1247,34 @@ const COORD TextBuffer::GetWordEnd(const COORD target, const std::wstring_view w
...
+    const auto bufferSize{ GetSize() };
+    COORD result{ target };
 
-    // Check if we're already on/past the last RegularChar
-    if (bufferSize.CompareInBounds(result, lastCharPos, true) >= 0)
+    if (bufferSize.CompareInBounds(target, lastCharPos, true) >= 0)
     {
-        return bufferSize.EndExclusive();
-    }
+        // if we're already on/past the last RegularChar,
```
Comment (codeofdusk) [inline src/buffer/out/textBuffer.cpp:1255]: Not quite the last regular char. We need (user-entered) spaces, just not the extraneous thousands of newlines.

**Comment by DHowett** [comment]
@carlos-zamora automerge broke this

**Comment by carlos-zamora** [comment]
> @carlos-zamora automerge broke this

Urgh. I rebased it locally, but haven't pushed the changes yet. I'll fix this on Monday. Thanks.

**Pull request closed without merging**